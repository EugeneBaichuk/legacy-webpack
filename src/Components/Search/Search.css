.search-scroll {
  height: 350px; 
  overflow-y: scroll; 
}
  
::-webkit-scrollbar {
  width: 10px;
}
  
::-webkit-scrollbar-track {
  border-radius: 3px;
}
   
::-webkit-scrollbar-thumb {
  background: #888; 
  border-radius: 3px;
}
  
::-webkit-scrollbar-thumb:hover {
  background: #555; 
}