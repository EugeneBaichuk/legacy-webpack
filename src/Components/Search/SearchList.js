import React from 'react'
import { ListGroup, ListGroupItem } from 'reactstrap';

const SearchList = ({filteredList}) => (
  <ListGroup className="mt-3 mb-3 search-scroll border">
    {filteredList?.map((result, i) => <ListGroupItem key={i}>{result}</ListGroupItem>)}
  </ListGroup>
)

export default SearchList
