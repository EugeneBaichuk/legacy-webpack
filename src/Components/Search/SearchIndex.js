import React, { useEffect, useState } from 'react';
import { Input } from 'reactstrap';
import SearchList from './SearchList';

import './Search.css';

const SearchIndex = ({ list }) => {
  const [searchInputValue, setSearchInputValue] = useState('');
  const [filteredList, setFilteredList] = useState(list);

  const handleInputChange = (e) => {
    setSearchInputValue(e.target.value);
  }

  useEffect(() => {
    const filteredList = searchInputValue ? 
      list.filter(item => item.includes(searchInputValue)):
      list;
    setFilteredList(filteredList);
  }, [searchInputValue]);

  return (
    <>
      <Input 
        className="mt-3" 
        value={searchInputValue} 
        placeholder='Search Here' 
        onChange={handleInputChange} 
      />
      <h3 className="m-3">Results: </h3>
      <SearchList filteredList={filteredList} />
    </>
  )
}

export default SearchIndex;