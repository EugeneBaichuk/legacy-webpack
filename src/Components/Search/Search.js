import React from 'react';
import Layout from '../Common/Layout';
import SearchIndex from './SearchIndex';
import { searchHeader, searchTaskList } from '../../Config/search.constants';
import { searchList } from '../../Config/search.constants';

export const Search = () => (
  <Layout 
    header={searchHeader} 
    list={searchTaskList}  
  >
    <SearchIndex list={searchList} />
  </Layout>
)
