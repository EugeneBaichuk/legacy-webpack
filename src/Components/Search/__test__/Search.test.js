import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Search } from '../Search';

describe('Cats component', () => {
  afterEach(cleanup);

  test('renders the header correctly', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment(<Search />)).toMatchSnapshot();
  });
});