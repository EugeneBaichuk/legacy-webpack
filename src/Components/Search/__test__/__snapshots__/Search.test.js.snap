// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cats component renders the header correctly 1`] = `
<DocumentFragment>
  <div
    class="layout container"
  >
    <h1
      class="mb-4 mt-4"
    >
      In this challenge your goal is to use the existing code to create the following:
    </h1>
    <ul
      class="list-group"
    >
      <li
        class="list-group-item"
      >
        ability to accept a search term from a user
      </li>
      <li
        class="list-group-item"
      >
        ability to filter the array in the state by that term
      </li>
      <li
        class="list-group-item"
      >
        ability to display only the terms that match the search, or all if no search term is provided
      </li>
    </ul>
    <hr
      class="my-4"
    />
    <input
      class="mt-3 form-control"
      placeholder="Search Here"
      type="text"
      value=""
    />
    <h3
      class="m-3"
    >
      Results: 
    </h3>
    <ul
      class="mt-3 mb-3 search-scroll border list-group"
    >
      <li
        class="list-group-item"
      >
        pen
      </li>
      <li
        class="list-group-item"
      >
        marker
      </li>
      <li
        class="list-group-item"
      >
        eraser
      </li>
      <li
        class="list-group-item"
      >
        notebook
      </li>
      <li
        class="list-group-item"
      >
        pencil
      </li>
      <li
        class="list-group-item"
      >
        scissors
      </li>
      <li
        class="list-group-item"
      >
        highlighter
      </li>
      <li
        class="list-group-item"
      >
        stapler
      </li>
      <li
        class="list-group-item"
      >
        paper clip
      </li>
      <li
        class="list-group-item"
      >
        binder
      </li>
      <li
        class="list-group-item"
      >
        hole punch
      </li>
      <li
        class="list-group-item"
      >
        laminator
      </li>
      <li
        class="list-group-item"
      >
        laminating sheets
      </li>
      <li
        class="list-group-item"
      >
        protective sheets
      </li>
      <li
        class="list-group-item"
      >
        index cards
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
